File: content/snippets/python/exercise_tree_traversal.py
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

def inorder(root, res):
    if not root:
        return
    inorder(root.left, res)
    res.append(root.val)
    inorder(root.right, res)

root = Node(1)
root.left = Node(2)
root.right = Node(3)
root.left.left = Node(4)

res = []
inorder(root, res)
print(res)

File: content/snippets/cpp/exercise_tree_traversal.cpp
#include <iostream>
#include <vector>

struct Node {
    int val;
    Node* left;
    Node* right;
    Node(int v) : val(v), left(nullptr), right(nullptr) {}
};

void inorder(Node* root, std::vector<int>& res) {
    if (!root) return;
    inorder(root->left, res);
    res.push_back(root->val);
    inorder(root->right, res);
}

int main() {
    Node* root = new Node(1);
    root->left = new Node(2);
    root->right = new Node(3);
    root->left->left = new Node(4);
    
    std::vector<int> res;
    inorder(root, res);
    
    for (int x : res) std::cout << x << " ";
    std::cout << "\n";
    return 0;
}
